### PR TITLE
Updated to Jupyter protocol v5

### LINKED ISSUE #44: Support for Jupyter Lab
I could successfully install ICSharp in Jupyter notebook in both Windows 10 Home and Linux docker container and able to run the sample notebook ([iCSharp Sample.ipynb](https://github.com/zabirauf/icsharp/blob/master/iPythonNotebook/iCSharp%20Sample.ipynb)).

After I installed [Jupyter Lab](https://jupyterlab.readthedocs.io/en/stable/), I can see that it somehow recognizes the kernel, because there is an option to start a notebook with C# kernel. When I select that option, the kernel json file content is display in the console:

```
2018/06/15 16:48:08:414 [DEBUG] Hello2
2018/06/15 16:48:08:421 [DEBUG] C:\Users\Alex\AppData\Roaming\jupyter\runtime\kernel-b9abf7cf-2f77-4b6f-973e-76ee2344f684.json
2018/06/15 16:48:08:428 [INFO]  Opening file C:\Users\Alex\AppData\Roaming\jupyter\runtime\kernel-b9abf7cf-2f77-4b6f-973e-76ee2344f684.json
2018/06/15 16:48:08:428 [DEBUG] {
  "shell_port": 64747,
  "iopub_port": 64748,
  "stdin_port": 64749,
  "control_port": 64750,
  "hb_port": 64751,
  "ip": "127.0.0.1",
  "key": "1bda1ee9-8192f31157878617edb9fca3",
  "transport": "tcp",
  "signature_scheme": "hmac-sha256",
  "kernel_name": "csharp"
}
```
shot

What changes should we make so that jupyter lab cell evaluate C# code?

### PR DESCRIPTION
**Pull request by wildart**
Update to Jupyter message specification v5 which allows to run the kernel on recent version of Jupyter Lab/Notebook. Closes #44.

### CODE REVIEW

**Comment by zabirauf** [comment]
Thanks a lot for the PR. I’ll take a look at it and merge it.

**Comment by wildart** [comment]
It looks like Travis chokes on dependency tests not related to this PR.

File: Kernel/Helpers/SignatureValidator.cs [hunk 1]
```diff
@@ -55,19 +55,6 @@ public string CreateSignature (Message message)
 			return BitConverter.ToString(this._signatureGenerator.Hash).Replace("-","").ToLower();
 		}
 
-		/// <summary>
-		/// Determines whether this instance is valid signature the specified message.
-		/// </summary>
-		/// <returns>true</returns>
-		/// <c>false</c>
-		/// <param name="message">Message.</param>
-		public bool IsValidSignature (Message message)
```
Comment (zabirauf) [inline Kernel/Helpers/SignatureValidator.cs:64]: Why is this removed?

File: Kernel/ScriptEngine/IoPubConsole.cs [hunk 2]
```diff
@@ -81,7 +82,7 @@ public void CancelRedirect()
 
         private void SendOutputMessageToIOPub(string value)
         {
-            Dictionary<string, object> data = new Dictionary<string, object>()
```
Comment (zabirauf) [inline Kernel/ScriptEngine/IoPubConsole.cs:84]: Just wondering whats the reason for changing it.

File: Kernel/Shell/ExecuteRequestHandler.cs [hunk 3]
```diff
@@ -54,33 +59,50 @@ public void HandleMessage(Message message, RouterSocket serverSocket, PublisherS
...
+                }
+
+                // 5: Send execute reply to shell socket
+                this.SendExecuteReplyOkMessage(message, serverSocket);
+            }
+            else
             {
-                this.SendOutputMessageToIOPub(message, ioPub, displayData);
+                var ex = results.CompileError != null ? results.CompileError : results.ExecuteError;
+                dynamic errorContent = new JObject();
```
Comment (zabirauf) [inline Kernel/Shell/ExecuteRequestHandler.cs:89]: Dont use dynamics, instead do the following
```
JObject errorContent = new JObject
{
  {"executiion_count", this.executionCount},
  {"ename", ex.GetType().ToString()},
  {"evalue", ex.Message}
};
```

File: Kernel/Shell/ExecuteRequestHandler.cs [hunk 4]
```diff
@@ -54,33 +59,50 @@ public void HandleMessage(Message message, RouterSocket serverSocket, PublisherS
...
+                    this.SendDisplayDataMessageToIOPub(message, ioPub, displayData);
+                }
+
+                // 5: Send execute reply to shell socket
+                this.SendExecuteReplyOkMessage(message, serverSocket);
+            }
+            else
             {
-                this.SendOutputMessageToIOPub(message, ioPub, displayData);
+                var ex = results.CompileError != null ? results.CompileError : results.ExecuteError;
```
Comment (zabirauf) [inline Kernel/Shell/ExecuteRequestHandler.cs:88]: Can you put this in a separate function?

File: Messages/Message.cs [hunk 5]
```diff
@@ -4,26 +4,25 @@ namespace iCSharp.Messages
...
-        [JsonProperty("identifiers")]
+        [JsonIgnoreAttribute]
         public List<byte[]> Identifiers { get; set; }
 
-        [JsonProperty("uuid")]
-        public string UUID { get; set; }
-
-        [JsonProperty("hmac")]
-        public string HMac { get; set; }
+        [JsonIgnoreAttribute]
```
Comment (zabirauf) [inline Messages/Message.cs:24]: We are not sending signature any more?

**Review by zabirauf — CHANGES_REQUESTED** [review]

**Pull request merged**